### ISSUE TITLE
Custom CUDA_ARCH doesn't seem to work

### ISSUE THREAD

**Issue by Spudz76** [fireice-uk/xmr-stak#1297]
Compiling dev branch with custom supplied CUDA_ARCH list does not run on targeted archs...
```
[2018-04-04 11:51:39] : Mining coin: karbo
CUDA [8.0] found GPU#0 arch [52], name [GeForce GTX 970]
WARNING: NVIDIA GPU 0: miner not compiled for the gpu architecture 52.
[2018-04-04 11:51:39] : WARNING: NVIDIA setup failed for GPU 0.

[2018-04-04 11:51:39] : NVIDIA: GPU configuration stored in file 'nvidia.txt'
[2018-04-04 11:51:39] : WARNING: backend NVIDIA disabled.
```

I added the CUDA detection report line as such:
```
--- a/xmrstak/backend/nvidia/nvcc_code/cuda_extra.cu
+++ b/xmrstak/backend/nvidia/nvcc_code/cuda_extra.cu
@@ -450,6 +450,9 @@ extern "C" int cuda_get_deviceinfo(nvid_ctx* ctx)

        ctx->name = std::string(props.name);

+    printf("CUDA [%d.%d] found GPU#%d arch [%d], name [%s]\n",
+           CUDART_VERSION / 1000, (CUDART_VERSION % 1000) / 10, ctx->device_id, gpuArch, ctx->device_name);
+
        std::vector<int> arch;
 #define XMRSTAK_PP_TOSTRING1(str) #str
 #define XMRSTAK_PP_TOSTRING(str) XMRSTAK_PP_TOSTRING1(str)
```

# Basic information
  - Type of the CPU.
Multiple / any (all show same issue)
  - Type of the GPU (if you try to miner with the GPU).
GTX550Ti, GTX770, GTX970, GTX1060 (any CUDA target it seems)
# Compile issues
  - Which OS do you use?
Both.  But posting from a Windows 7 compile attempt, same issue exists on Linux done the same way
  ```
set PATH=C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v8.0\bin;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v8.0\libnvvp;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files (x86)\AMD APP SDK\3.0\bin\x86_64;C:\Program Files (x86)\AMD APP SDK\3.0\bin\x86;C:\Program Files\CMake\bin;C:\Program Files\Git\cmd;C:\Program Files\Redis\;C:\Program Files\PuTTY\;C:\Program Files (x86)\Nodist\bin;C:\Program Files\dotnet\;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\Wbem;C:\Windows\system32;C:\Windows
set CUDA_PATH=C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v8.0
set NVCUDASAMPLES_ROOT=C:\ProgramData\NVIDIA Corporation\CUDA Samples\v8.0
set CMAKE_PREFIX_PATH=..\xmr-stak-dep_v1\hwloc;..\xmr-stak-dep_v1\libmicrohttpd;..\xmr-stak-dep_v1\openssl
call "C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\Tools\VsMSBuildCmd.bat"
call "C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Auxiliary\Build\vcvarsall.bat" x64 -vcvars_ver=14.0
cmake -G "Visual Studio 15 2017 Win64" -T v140,host=x64 -DCMAKE_BUILD_TYPE=Release -DCUDA_ARCH="20;30;52;61" -DCUDA_SHOW_REGISTER=ON -DXMR-STAK_LARGEGRID=ON ..
cmake --build . --config Release --target clean
cmake --build . --config Release --target install
  ```
  ```
-- Configuring done
-- Generating done
-- Build files have been written to: C:/Users/AccountName/CLionProjects/xmr-stak/build-80
-- Cache values
CMAKE_CONFIGURATION_TYPES:STRING=Debug;Release;MinSizeRel;RelWithDebInfo
CMAKE_CXX_FLAGS:STRING=/DWIN32 /D_WINDOWS /W3 /GR /EHsc
CMAKE_CXX_FLAGS_DEBUG:STRING=/MDd /Zi /Ob0 /Od /RTC1
CMAKE_CXX_FLAGS_MINSIZEREL:STRING=/MD /O1 /Ob1 /DNDEBUG
CMAKE_CXX_FLAGS_RELEASE:STRING=/MD /O2 /Ob2 /DNDEBUG
CMAKE_CXX_FLAGS_RELWITHDEBINFO:STRING=/MD /Zi /O2 /Ob1 /DNDEBUG
CMAKE_CXX_STANDARD_LIBRARIES:STRING=kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib
CMAKE_C_FLAGS:STRING=/DWIN32 /D_WINDOWS /W3
CMAKE_C_FLAGS_DEBUG:STRING=/MDd /Zi /Ob0 /Od /RTC1
CMAKE_C_FLAGS_MINSIZEREL:STRING=/MD /O1 /Ob1 /DNDEBUG
CMAKE_C_FLAGS_RELEASE:STRING=/MD /O2 /Ob2 /DNDEBUG
CMAKE_C_FLAGS_RELWITHDEBINFO:STRING=/MD /Zi /O2 /Ob1 /DNDEBUG
CMAKE_C_STANDARD_LIBRARIES:STRING=kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib
CMAKE_EXE_LINKER_FLAGS:STRING=/machine:x64
CMAKE_EXE_LINKER_FLAGS_DEBUG:STRING=/debug /INCREMENTAL
CMAKE_EXE_LINKER_FLAGS_MINSIZEREL:STRING=/INCREMENTAL:NO
CMAKE_EXE_LINKER_FLAGS_RELEASE:STRING=/INCREMENTAL:NO
CMAKE_EXE_LINKER_FLAGS_RELWITHDEBINFO:STRING=/debug /INCREMENTAL
CMAKE_INSTALL_PREFIX:PATH=C:/Users/AccountName/CLionProjects/xmr-stak/build-80
CMAKE_LINKER:FILEPATH=C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/amd64/link.exe
CMAKE_LINK_STATIC:BOOL=OFF
CMAKE_MODULE_LINKER_FLAGS:STRING=/machine:x64
CMAKE_MODULE_LINKER_FLAGS_DEBUG:STRING=/debug /INCREMENTAL
CMAKE_MODULE_LINKER_FLAGS_MINSIZEREL:STRING=/INCREMENTAL:NO
CMAKE_MODULE_LINKER_FLAGS_RELEASE:STRING=/INCREMENTAL:NO
CMAKE_MODULE_LINKER_FLAGS_RELWITHDEBINFO:STRING=/debug /INCREMENTAL
CMAKE_RC_COMPILER:FILEPATH=C:/Program Files (x86)/Windows Kits/10/bin/10.0.16299.0/x64/rc.exe
CMAKE_RC_FLAGS:STRING=/DWIN32
CMAKE_RC_FLAGS_DEBUG:STRING=/D_DEBUG
CMAKE_RC_FLAGS_MINSIZEREL:STRING=
CMAKE_RC_FLAGS_RELEASE:STRING=
CMAKE_RC_FLAGS_RELWITHDEBINFO:STRING=
CMAKE_SHARED_LINKER_FLAGS:STRING=/machine:x64
CMAKE_SHARED_LINKER_FLAGS_DEBUG:STRING=/debug /INCREMENTAL
CMAKE_SHARED_LINKER_FLAGS_MINSIZEREL:STRING=/INCREMENTAL:NO
CMAKE_SHARED_LINKER_FLAGS_RELEASE:STRING=/INCREMENTAL:NO
CMAKE_SHARED_LINKER_FLAGS_RELWITHDEBINFO:STRING=/debug /INCREMENTAL
CMAKE_SKIP_INSTALL_RPATH:BOOL=NO
CMAKE_SKIP_RPATH:BOOL=NO
CMAKE_STATIC_LINKER_FLAGS:STRING=/machine:x64
CMAKE_STATIC_LINKER_FLAGS_DEBUG:STRING=
CMAKE_STATIC_LINKER_FLAGS_MINSIZEREL:STRING=
CMAKE_STATIC_LINKER_FLAGS_RELEASE:STRING=
CMAKE_STATIC_LINKER_FLAGS_RELWITHDEBINFO:STRING=
CMAKE_VERBOSE_MAKEFILE:BOOL=FALSE
CPU_ENABLE:BOOL=ON
CUDA_64_BIT_DEVICE_CODE:BOOL=ON
CUDA_ARCH:STRING=20;21;30;52;61
CUDA_ATTACH_VS_BUILD_RULE_TO_CUDA_FILE:BOOL=ON
CUDA_BUILD_CUBIN:BOOL=OFF
CUDA_BUILD_EMULATION:BOOL=OFF
CUDA_COMPILER:STRING=nvcc
CUDA_CUDART_LIBRARY:FILEPATH=C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v8.0/lib/x64/cudart.lib
CUDA_CUDA_LIBRARY:FILEPATH=C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v8.0/lib/x64/cuda.lib
CUDA_ENABLE:BOOL=ON
CUDA_GENERATED_OUTPUT_DIR:PATH=
CUDA_HOST_COMPILATION_CPP:BOOL=ON
CUDA_HOST_COMPILER:FILEPATH=$(VCInstallDir)bin
CUDA_KEEP_FILES:BOOL=OFF
CUDA_NVCC_EXECUTABLE:FILEPATH=C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v8.0/bin/nvcc.exe
CUDA_NVCC_FLAGS:STRING=
CUDA_NVCC_FLAGS_DEBUG:STRING=
CUDA_NVCC_FLAGS_MINSIZEREL:STRING=
CUDA_NVCC_FLAGS_RELEASE:STRING=
CUDA_NVCC_FLAGS_RELWITHDEBINFO:STRING=
CUDA_PROPAGATE_HOST_FLAGS:BOOL=ON
CUDA_SDK_ROOT_DIR:PATH=CUDA_SDK_ROOT_DIR-NOTFOUND
CUDA_SEPARABLE_COMPILATION:BOOL=OFF
CUDA_SHOW_CODELINES:BOOL=OFF
CUDA_SHOW_REGISTER:BOOL=ON
CUDA_TOOLKIT_INCLUDE:PATH=C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v8.0/include
CUDA_TOOLKIT_ROOT_DIR:PATH=C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v8.0
CUDA_USE_STATIC_CUDA_RUNTIME:BOOL=ON
CUDA_VERBOSE_BUILD:BOOL=OFF
CUDA_VERSION:STRING=8.0
CUDA_cublas_LIBRARY:FILEPATH=C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v8.0/lib/x64/cublas.lib
CUDA_cublas_device_LIBRARY:FILEPATH=C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v8.0/lib/x64/cublas_device.lib
CUDA_cudadevrt_LIBRARY:FILEPATH=C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v8.0/lib/x64/cudadevrt.lib
CUDA_cudart_static_LIBRARY:FILEPATH=C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v8.0/lib/x64/cudart_static.lib
CUDA_cufft_LIBRARY:FILEPATH=C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v8.0/lib/x64/cufft.lib
CUDA_cupti_LIBRARY:FILEPATH=C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v8.0/extras/CUPTI/libx64/cupti.lib
CUDA_curand_LIBRARY:FILEPATH=C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v8.0/lib/x64/curand.lib
CUDA_cusolver_LIBRARY:FILEPATH=C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v8.0/lib/x64/cusolver.lib
CUDA_cusparse_LIBRARY:FILEPATH=C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v8.0/lib/x64/cusparse.lib
CUDA_nppc_LIBRARY:FILEPATH=C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v8.0/lib/x64/nppc.lib
CUDA_nppi_LIBRARY:FILEPATH=C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v8.0/lib/x64/nppi.lib
CUDA_npps_LIBRARY:FILEPATH=C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v8.0/lib/x64/npps.lib
CUDA_nvcuvenc_LIBRARY:FILEPATH=CUDA_nvcuvenc_LIBRARY-NOTFOUND
CUDA_nvcuvid_LIBRARY:FILEPATH=C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v8.0/lib/x64/nvcuvid.lib
HWLOC:FILEPATH=C:/Users/AccountName/CLionProjects/xmr-stak/xmr-stak-dep_v1/hwloc/lib/libhwloc.lib
HWLOC_ENABLE:BOOL=ON
HWLOC_INCLUDE_DIR:PATH=C:/Users/AccountName/CLionProjects/xmr-stak/xmr-stak-dep_v1/hwloc/include
LIB_EAY_DEBUG:FILEPATH=C:/OpenSSL-Win64/lib/VC/libeay32MDd.lib
LIB_EAY_RELEASE:FILEPATH=C:/Users/AccountName/CLionProjects/xmr-stak/xmr-stak-dep_v1/openssl/lib/libeay32.lib
MHTD:FILEPATH=C:/Users/AccountName/CLionProjects/xmr-stak/xmr-stak-dep_v1/libmicrohttpd/lib/libmicrohttpd.lib
MICROHTTPD_ENABLE:BOOL=ON
MTHD_INCLUDE_DIR:PATH=C:/Users/AccountName/CLionProjects/xmr-stak/xmr-stak-dep_v1/libmicrohttpd/include
OPENSSL_INCLUDE_DIR:PATH=C:/Users/AccountName/CLionProjects/xmr-stak/xmr-stak-dep_v1/openssl/include
OpenCL_ENABLE:BOOL=ON
OpenCL_INCLUDE_DIR:PATH=C:/Program Files (x86)/AMD APP SDK/3.0/include
OpenCL_LIBRARY:FILEPATH=C:/Program Files (x86)/AMD APP SDK/3.0/lib/x86_64/OpenCL.lib
OpenSSL_ENABLE:BOOL=ON
SSL_EAY_DEBUG:FILEPATH=C:/OpenSSL-Win64/lib/VC/ssleay32MDd.lib
SSL_EAY_RELEASE:FILEPATH=C:/Users/AccountName/CLionProjects/xmr-stak/xmr-stak-dep_v1/openssl/lib/ssleay32.lib
XMR-STAK_COMPILE:STRING=native
XMR-STAK_LARGEGRID:BOOL=ON
XMR-STAK_THREADS:STRING=0
  ```

# Issue with the execution
  - Do you compiled the miner by our own?
Of course!  From dev branch otherwise no fun!
  ```
Version: xmr-stak/2.4.1/5ce9892/dev/win/nvidia-amd-cpu/aeon-cryptonight-monero/0
  ```

# Stability issue
It never starts using the GPU as it fails to agree it contains support for the architecture.

**Comment by psychocrypt** [comment]
please remove `"` from `CUDA_ARCH`. It is a cmake list therefore `"` is not allowed. It must be `-DCUDA_ARCH=20;30;52;61 `

**Comment by psychocrypt** [comment]
I found that it is a bug if architecture 20 is mixed with other architectures. It is solved with #1299

**Comment by Spudz76** [comment]
Strange, the register dump shows it building everything as requested... quotes or no

It has to be in quotes in Linux or else it means ; next command in a shell?
